### PR TITLE
[HZ-1240] Fix incorrect conversion to select-by-key-rel

### PR DESCRIPTION
**Pull request by Fly-Style**
This PR fixes the behavior when we have optimized single-key read from HZ IMDG rel (SelectByKeyMapPhysicalRel) **not** as a root rel in the rel tree, as it is expected to be. Instead, we'll deopt this rel to a simple `full scan` with the condition.

### CODE REVIEW

**Comment by Fly-Style** [comment]
run-lab-run

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/SelectByKeyDeoptPhysicalRule.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.calcite.rel.RelNode;
+import org.immutables.value.Value;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.sql.impl.opt.Conventions.PHYSICAL;
+
+@Value.Enclosing
+final class SelectByKeyDeoptPhysicalRule extends RelRule<RelRule.Config> {
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/SelectByKeyDeoptPhysicalRule.java:33]: This is weird. Ideally we shouldn't have produced a non-working plan in the first place. We probably can't have a rule that will match an element on the root, but instead after logical opt we can look at the whole plan, and convert to select-by-key, if the whole tree is just a single full-scan.

**Review by viliam-durina — CHANGES_REQUESTED** [review]

**Review by viliam-durina — APPROVED** [review]

**Review by krzysztofslusarski — APPROVED** [review]

**Comment by Fly-Style** [comment]
run-lab-run

**Comment by Fly-Style** [comment]
run-ee-compiler

**Comment by Fly-Style** [comment]
run-ee-compile

**Pull request merged**